### PR TITLE
Better error handling in infoview

### PR DESCRIPTION
**Pull request by gebner**
The Lean 3 widget implementation had some issues (tried to serialize a lua function as json).  This caused neovim to raise an exception, which was not handled.  Due to the exception the lock in the tick remained locked and prevented further updates, causing the infoview to hang.

This PR fixes the function serialization issue and disables the locking in the ticks because I didn't find a reliable way to handle these fatal errors.

### CODE REVIEW

**Comment by Julian** [comment]
Two more checked off from #139 sounds like :) -- thanks again! LGTM.

File: lua/lean/lean3.lua [hunk 1]
```diff
@@ -215,7 +215,8 @@ function lean3.update_infoview(pin, bufnr, params, use_widget, opts, options)
           local clickable_event = div_event == "click" or div_event == "change"
           if clickable_event then element_div.highlightable = true end
           events[div_event] = function(this_tick, value)
```
Comment (rish987) [inline lua/lean/lean3.lua:217]: My bad! I should have made `value` the third parameter after I updated to be passing in the `render_fn`. Though this fixes it as well, right? So could we revert 2589253? (again, I'm a bit occupied this week so I may not be able to do much more that comment)

File: lua/lean/_util.lua [hunk 2]
```diff
@@ -107,6 +107,9 @@ end
 
 M.a_request = a.wrap(M.request, 4)
 
+-- FIXME: tick locking is disabled for now
```
Comment (rish987) [inline lua/lean/_util.lua:110]: I think a better alternative to this is using a variant of `lsp.buf_request` that has a built-in timer after which it expires and calls the handler with an error (not sure if this already exists).

File: lua/lean/lean3.lua (hunk 1 above)
Comment (rish987) [reply lua/lean/lean3.lua:217]: (specifically, this was introduced in 27977d1)

File: lua/lean/_util.lua (hunk 2 above)
Comment (gebner) [reply lua/lean/_util.lua:110]: That's what I tried first, but I couldn't get it working (without spending too much time on it).  For now, I think it's less annoying if the infoview is sometimes corrupted than if it stops updating completely.

Comment (rish987) [reply lua/lean/_util.lua:110]: Right, but doesn't 6d8314c fix it without this?

Comment (gebner) [reply lua/lean/_util.lua:110]: I've run into similar problems (infoview not updating) with Lean 4 as well, but they're harder to reproduce.

Comment (rish987) [reply lua/lean/_util.lua:110]: Interesting. And you're sure the the Lean 4 server is guaranteed to respond to every LSP request that we're making here? And when it did halt in Lean 4, did you get an error?

I'm just concerned that doing this -- in addition to the bugs it introduces -- serves to hide issues that we need to solve, here or possibly even in the server itself.

**Pull request merged**